harvard-huit/Harvard-Geospatial-Library
# Fixing JS bug that attempted to create duplicate leaflet maps.

**Fixing JS bug that attempted to create duplicate leaflet maps.**
* * *

**JIRA Ticket**: none

# What does this Pull Request do?
* When viewing detail pages when the JavaScript is compressed (as it would be in QA and Production environments), there is a giant JavaScript error in the console ("Uncaught Error: Map container is already initialized.") and scripts below that error do not fire (For example, the titles under the "About this item" heading in the right rail do not display in the browser).
* I also updated the Test and Production environments to properly compress JS and SASS files.
* I also updated DockerfileLocal to properly use the RAILS_ENV

# How should this be tested?

* To reproduce the problem, temporarily switch your local environment to the Test environment (you do not need to fully rebuild).  To do this change the RAILS_ENV variable to test in your .env AND ALSO change passenger_app_env to test in your webapp.conf 
* After changing that setting, You can see the error in the console and the titles not load on a detail page.  Additionally if you click on a map tile the index explorer will open but you will not be able to close it. This detail page is a good example: https://localhost:3001/catalog/harvard-g5831-r3-s20-f7-index-gr
* After incorporating this PR / Branch you can restart your container (you do not need to fully rebuild) then refresh the page. The JS error in the console will disappear and the rest of the JS actions on the page will work properly.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No
- integration tests? No

# Interested parties
@mferrarini @enriquediaz
